.github/workflows/unit-testing.yml

```diff
@@ -0,0 +1,89 @@
+name: Unit Testing
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  unit-testing-core:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: npm clean install
+        run: npm ci
+        working-directory: ./GenForm-NPM/Core/
+
+      - name: npm run tests for core
+        run: npm test
+        working-directory: ./GenForm-NPM/Core/
+
+  cypress-react:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: npm install in Exemples
+        run: npm install
+        working-directory: ./GenForm-Exemples/React/
+
+      - name: install cypress in Exemples
+        run: npm install cypress
+        working-directory: ./GenForm-Exemples/React/
+
+      - name: run project
+        run: (npm run start --port 3000&)
+        working-directory: ./GenForm-Exemples/React/
+
+      - name: npm install in Tests
+        run: npm install
+        working-directory: ./GenForm-Tests/Cypress/
+
+      - name: install cypress in Tests
```

Comment:
Pas nécessaire

Reply:
j'ai besoin d'installer cypress pour l'utiliser non ?

Reply:
Justement non, vu que le cypress tu le lances depuis le dossier GenForm-Tests/Cypress

Reply:
oui mais là la ligne 52 c'est l'installation de cypress dans ./GenForm-Tests/Cypress/

Reply:
Bah chelou c'est dupliqué, du genre la step d'en haut et cette step

Reply:
je comprends plus là
![image](https://github.com/GenForm/GenForm/assets/60664665/4ddfc4d7-7251-490c-b599-39f8ef4deb36)

Reply:
Si on prend là "pipeline" pour React : 
![image](https://github.com/GenForm/GenForm/assets/73128006/a11ee3e2-471a-470a-a5fb-5510490ff50e)
La step "npm install in Tests" et "install Cypress in Tests" ont la même fonction, installer cypress (vu que cypress est contenu dans package.json)

Reply:
ahhhhh